backend/main/src/Group/Query/FindUsersBelongingToAGroupQuery/FindUsersBelongingToAGroupQuery.ts

```diff
@@ -0,0 +1,51 @@
+import { IQuery } from "src/Shared/Layers/Query/IQuery";
+import { NishikiDynamoDBClient } from "src/Shared/Adapters/DB/NishikiTableClient";
+import { Err, Ok, Result } from "result-ts-type";
+import { QueryError } from "src/Shared/Utils/Errors";
+import { isValidUUIDV4 } from "src/Shared/Utils/Validator";
+
+export class FindUsersBelongingToAGroupQuery
+	implements IQuery<{ groupId: string }, IUser[], InvalidUUIDV4>
+{
+	constructor(public readonly nishikiDynamoDBClient: NishikiDynamoDBClient) {}
+
+	public async execute(input: { groupId: string }): Promise<
+		Result<IUser[], InvalidUUIDV4>
+	> {
+		const { groupId } = input;
+
+		if (!isValidUUIDV4(groupId)) {
+			return Err(new InvalidUUIDV4());
+		}
+
+		const result = await this.nishikiDynamoDBClient.listOfUsersInGroup(groupId);
+
+		if (result.length === 0) return Ok([]);
+
+		const usersData = await Promise.all(
+			result.map((user) =>
+				this.nishikiDynamoDBClient.getUser({ userId: user.userId }),
+			),
+		);
+
+		const users: IUser[] = [];
+
+		for (const userData of usersData) {
+			if (userData) {
+				users.push({
+					id: userData.userId,
+					name: userData.username,
+				});
+			}
+		}
+
+		return Ok(users);
```

Comment:
This comment is for confirmation.
Based on the API doc, the final data format should be `return Ok({users: users});`. But If my understanding is correct, we will format data in the router. Is that right?

Reply:
No. Your understanding is incorrect.
This one comes from my overlooking.

I've fixed it. Thank you for pointing it out.

Reply:
Thank you. I misunderstood it. Now I got it.